routes/donations.js

```diff
@@ -9,11 +9,19 @@ const {
 const { getUser } = require('../models/users');
 
 router.get('/', (req, res) => {
-    res.status(200).json(getDonationsForIdea(req.query.ideaId));
+    try {
+        res.status(200).json(getDonationsForIdea(req.query.ideaId));
+    } catch (error) {
+        res.status(404).send(error.message);
+    }
 });
 
 router.get('/total', (req, res) => {
-    res.status(200).json(getTotalSumDonatedForIdea(req.query.ideaId));
+    try {
+        res.status(200).json(getTotalSumDonatedForIdea(req.query.ideaId));
+    } catch (error) {
+        res.status(404).send(error.message);
```

Comment:
If unexpected server error happens, then server should return HTTP 500, not 404

Reply:
Phind fixed.